### ISSUE TITLE
CRC start needs admin permissions mid script

### ISSUE THREAD

**Issue by jlynchMicron** [code-ready/crc#1175]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version
```bash
# Put the output of `crc version`
crc version: 1.8.0+0a318dc
OpenShift version: 4.3.8 (embedded in binary)
```
  
## CRC status
```bash
# Put the output of `crc status`
ERRO error: CreateFile C:\Users\jlynch\.crc\machines\crc\kubeconfig: The system cannot find the file specified.
 - exit status 1
```

## CRC config
```bash
# Put the output of `crc config view`
- http-proxy                            : ### corp proxy ###
- https-proxy                           : ### corp proxy ###
```

## Host Operating System
```bash
Host Name:                 JLYNCH10
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Micron
Registered Organization:   Micron Technology, Inc.
Product ID:                00329-00000-00003-AA175
Original Install Date:     11/12/2019, 12:22:10 PM
System Boot Time:          4/20/2020, 2:26:03 PM
System Manufacturer:       HP
System Model:              HP EliteDesk 800 G3 SFF
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 94 Stepping 3 GenuineIntel ~3192 Mhz
BIOS Version:              HP P01 Ver. 02.31, 10/15/2019
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume2
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-07:00) Mountain Time (US & Canada)
Total Physical Memory:     32,647 MB
Available Physical Memory: 17,104 MB
Virtual Memory: Max Size:  37,511 MB
Virtual Memory: Available: 20,028 MB
Virtual Memory: In Use:    17,483 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    na.micron.com
Logon Server:              \\WDCNABO3
Hotfix(s):                 11 Hotfix(s) Installed.
                           [01]: KB4537490
                           [02]: KB4465065
                           [03]: KB4470788
                           [04]: KB4489907
                           [05]: KB4494174
                           [06]: KB4521862
                           [07]: KB4523204
                           [08]: KB4537759
                           [09]: KB4539571
                           [10]: KB4549947
                           [11]: KB4538461
Network Card(s):           2 NIC(s) Installed.
                           [01]: Intel(R) Ethernet Connection (5) I219-LM
                                 Connection Name: Ethernet
                                 DHCP Enabled:    Yes
                                 DHCP Server:     137.201.91.43
                                 IP address(es)
                                 [01]: 137.201.45.33
                                 [02]: fe80::ad41:64ba:f17b:2cf8
                           [02]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (Default Switch)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 192.168.202.81
                                 [02]: fe80::1532:b455:f0d0:be27
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. crc setup
  2. crc start -p C:\Users\jlynch\Programs\crc-windows-amd64\crc-windows-1.8.0-amd64\pull-secret.txt --log-level debug

### Expected
Successfully start crc in Hyper-V.

### Actual
Script needs admin permission in middle of script which I cannot elevate in due to corporate antivirus block or permissions issue.  I can either start powershell as admin or normal user, past that, I do not know how to elevate permissions once powershell is started as normal user.

`INFO Will run as admin: add dns server address to interface vEthernet (Default Switch)`

### Logs

```bash
PS C:\Users\jlynch\Programs\crc-windows-amd64\crc-windows-1.8.0-amd64> crc start -p C:\Users\jlynch\Programs\crc-windows-amd64\crc-windows-1.8.0-amd64\pull-secret.txt --log-level debug
DEBU HTTP-PROXY: http://proxy-web.micron.com:80, HTTPS-PROXY: http://proxy-web.micron.com:80, NO-PROXY: 127.0.0.1,localhost,localhost,127.0.0.1,.micron.com,addmmsi
DEBU No new version available. The latest version is 1.8.0
INFO Checking if oc binary is cached
DEBU oc binary already cached
INFO Checking if podman remote binary is cached
DEBU podman remote currently not supported on this platform
INFO Checking if running as normal user
INFO Checking Windows 10 release
INFO Checking if Hyper-V is installed and operational
INFO Checking if user is a member of the Hyper-V Administrators group
INFO Checking if Hyper-V service is enabled
INFO Checking if the Hyper-V virtual switch exist
INFO Found Virtual Switch to use: Default Switch
Checking file: C:\Users\jlynch\.crc\machines\crc\.crc-exist
INFO Loading bundle: crc_hyperv_4.3.8.crcbundle ...
INFO Checking size of the disk image C:\Users\jlynch\.crc\cache\crc_hyperv_4.3.8\crc.vhdx ...
INFO Creating CodeReady Containers VM for OpenShift 4.3.8...
Found binary path at C:\Users\jlynch\Programs\crc-windows-amd64\crc-windows-1.8.0-amd64\crc.exe
Launching plugin server for driver hyperv
Plugin server listening at address 127.0.0.1:51669
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(crc) Calling .GetMachineName
(crc) Calling .DriverName
Running pre-create checks...
(crc) Calling .PreCreateCheck
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive @(Get-Module -ListAvailable hyper-v).Name | Get-Unique
(crc) DBG | [stdout =====>] : Hyper-V
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive @([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole(([System.Security.Principal.SecurityIdentifier]::new("S-1-5-32-578")))
(crc) DBG | [stdout =====>] : True
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive [Console]::OutputEncoding = [Text.Encoding]::UTF8; (Hyper-V\Get-VMSwitch).Name
(crc) DBG | [stdout =====>] : Default Switch
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .GetConfigRaw
Creating machine...
(crc) Calling .Create
(crc) Downloading C:\Users\jlynch\.crc\cache\crc.vhdx from file://C:/Users/jlynch/.crc/cache/crc_hyperv_4.3.8/crc.vhdx...
(crc) Creating VM...
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive [Console]::OutputEncoding = [Text.Encoding]::UTF8; (Hyper-V\Get-VMSwitch).Name
(crc) DBG | [stdout =====>] : Default Switch
(crc) DBG |
(crc) Using switch "Default Switch"
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\New-VM crc -Path 'C:\Users\jlynch\.crc\machines\crc' -SwitchName 'Default Switch' -MemoryStartupBytes 8192MB
(crc) DBG | [stdout =====>] :
(crc) DBG | Name State CPUUsage(%) MemoryAssigned(M) Uptime   Status             Version
(crc) DBG | ---- ----- ----------- ----------------- ------   ------             -------
(crc) DBG | crc  Off   0           0                 00:00:00 Operating normally 9.0
(crc) DBG |
(crc) DBG |
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\Set-VMMemory -VMName crc -DynamicMemoryEnabled $false
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\Set-VMProcessor crc -Count 4
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\Add-VMHardDiskDrive -VMName crc -Path 'C:\Users\jlynch\.crc\machines\crc\crc.vhdx'
(crc) Starting VM...
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\Start-VM crc
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) Waiting for host to start...
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] :
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) Calling .GetConfigRaw
(crc) DBG | [stderr =====>] :
(crc) Calling .DriverName
(crc) Calling .DriverName
Waiting for machine to be running, this may take a few minutes...
(crc) Calling .GetState
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
Machine is up and running!
(crc) DBG | [stderr =====>] :
Machine successfully created
Created C:\Users\jlynch\.crc\machines\crc\.crc-exist
(crc) Calling .GetState
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
Found binary path at C:\Users\jlynch\Programs\crc-windows-amd64\crc-windows-1.8.0-amd64\crc.exe
Launching plugin server for driver hyperv
Plugin server listening at address 127.0.0.1:51695
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
DEBU Waiting until ssh is available
(crc) Calling .GetSSHHostname
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .GetSSHPort
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} core [0x7bb780] 0x7bb750 <nil>  [] 0s} 192.168.202.92 22 <nil> <nil>}
DEBU About to run SSH command:
exit 0
DEBU SSH cmd err, output: <nil>:
INFO Verifying validity of the cluster certificates ...
(crc) Calling .GetSSHHostname
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .GetSSHPort
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} core [0x7bb780] 0x7bb750 <nil>  [] 0s} 192.168.202.92 22 <nil> <nil>}
DEBU About to run SSH command:
date --date="$(sudo openssl x509 -in /var/lib/kubelet/pki/kubelet-client-current.pem -noout -enddate | cut -d= -f 2)" --iso-8601=seconds
DEBU SSH cmd err, output: <nil>: 2020-04-26T15:41:22+00:00
(crc) Calling .GetIP
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .DriverName
(crc) Calling .GetSSHHostname
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .GetSSHPort
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} core [0x7bb780] 0x7bb750 <nil>  [] 0s} 192.168.202.92 22 <nil> <nil>}
DEBU About to run SSH command:
echo 'dXNlcj1yb290CnBvcnQ9IDUzCmJpbmQtaW50ZXJmYWNlcwpleHBhbmQtaG9zdHMKbG9nLXF1ZXJpZXMKc3J2LWhvc3Q9X2V0Y2Qtc2VydmVyLXNzbC5fdGNwLmNyYy50ZXN0aW5nLGV0Y2QtMC5jcmMudGVzdGluZywyMzgwLDEwCmxvY2FsPS9jcmMudGVzdGluZy8KZG9tYWluPWNyYy50ZXN0aW5nCmFkZHJlc3M9L2FwcHMtY3JjLnRlc3RpbmcvMTkyLjE2OC4yMDIuOTIKYWRkcmVzcz0vZXRjZC0wLmNyYy50ZXN0aW5nLzE5Mi4xNjguMjAyLjkyCmFkZHJlc3M9L2FwaS5jcmMudGVzdGluZy8xOTIuMTY4LjIwMi45MgphZGRyZXNzPS9hcGktaW50LmNyYy50ZXN0aW5nLzE5Mi4xNjguMjAyLjkyCmFkZHJlc3M9L2NyYy00NW5zay1tYXN0ZXItMC5jcmMudGVzdGluZy8xOTIuMTY4LjIwMi45Mgo=' | openssl enc -base64 -d | sudo tee /var/srv/dnsmasq.conf > /dev/null
DEBU SSH cmd err, output: <nil>:
(crc) Calling .GetSSHHostname
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .GetSSHPort
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} core [0x7bb780] 0x7bb750 <nil>  [] 0s} 192.168.202.92 22 <nil> <nil>}
DEBU About to run SSH command:
sudo podman rm -f dnsmasq
DEBU SSH cmd err, output: Process exited with status 1: Error: Failed to evict container: "": Failed to find container "dnsmasq" in state: no container with name or ID dnsmasq found: no such container
(crc) Calling .GetSSHHostname
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .GetSSHPort
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} core [0x7bb780] 0x7bb750 <nil>  [] 0s} 192.168.202.92 22 <nil> <nil>}
DEBU About to run SSH command:
sudo rm -f /var/lib/cni/networks/podman/10.88.0.8
DEBU SSH cmd err, output: <nil>:
(crc) Calling .GetSSHHostname
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM crc ).state
(crc) DBG | [stdout =====>] : Running
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Hyper-V\Get-VM crc ).networkadapters[0]).ipaddresses[0]
(crc) DBG | [stdout =====>] : 192.168.202.92
(crc) DBG |
(crc) DBG | [stderr =====>] :
(crc) Calling .GetSSHPort
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHKeyPath
(crc) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} core [0x7bb780] 0x7bb750 <nil>  [] 0s} 192.168.202.92 22 <nil> <nil>}
DEBU About to run SSH command:
sudo podman run  --ip 10.88.0.8 --name dnsmasq -v /var/srv/dnsmasq.conf:/etc/dnsmasq.conf -p 53:53/udp --privileged -d quay.io/crcont/dnsmasq:latest
DEBU SSH cmd err, output: <nil>: a7a158e79bcb36f717fa446f31b04b2909f2e74e28a939a609e36413b8b627d2
INFO Will run as admin: add dns server address to interface vEthernet (Default Switch)
ERRO Nameserver not successfully set
ERRO Error running post start: Nameserver not successfully set
Making call to close driver server
(crc) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
Making call to close driver server
(crc) Calling .Close
(crc) DBG | Closing plugin on server side
Successfully made call to close driver server
Making call to close connection to plugin binary
(crc) DBG | Closing plugin on server side
PS C:\Users\jlynch\Programs\crc-windows-amd64\crc-windows-1.8.0-amd64>
```

**Comment by gbraad** [comment]
Invalid: this is needed to setup DNS configuration.

```
INFO Will run as admin: add dns server address to interface vEthernet (Default Switch)
ERRO Nameserver not successfully set
```

If not possible, you would need to set entries in the `hosts` file. Do you have permissions for this? If not, make sure an exception is made inside the anti-virus tool or permissions for this file are granted

**Comment by gbraad** [comment]
See:

  * Issue #1132 Use /etc/hosts instead of dnsmasq #1136
  * [spike] Remove dns dependency on the host #1132

**Comment by jlynchMicron** [comment]
I think I should have permissions to edit the "hosts" file within the crc VM as long as I know the root password for getting into the VM.  I don't think editing anything within the VM would conflict with my Windows permission issues.
I am a little confused on what I should be adding to the "hosts" file though based on the issues you linked.  What needs to be added or edited?

**Comment by gbraad** [comment]
> I think I should have permissions to edit the "hosts" file within the crc VM 

No, DNS is resolved from the host `C:\Windows\System32\Drivers\etc\hosts`. The VM uses a dns server that handles the patterns for crc.testing and crc-apps.testing.

**Comment by jlynchMicron** [comment]
@gbraad sorry for the delayed response.  I tried running `C:\Windows\System32\netsh.exe interface ip set dns "vEthernet (Default Switch)" static 192.168.210.250 primary` as an admin in a separate shell when I ran into this problem running "crc start" as a regular user, but I would get this error `Your computer is trying to use a DNS server that is incorrect or doesn't exist`.  At this point it is starting to look like too much trouble for me trying to get this set up on a corporate Windows machine.  I think I will try this again on one of our Linux boxes that I can get admin permissions on to see if I have any more luck with that.

**Comment by jlynchMicron** [comment]
Hi @gbraad , I have not been able to get my hands on a Linux box yet, but I did try to install crc again with version 1.13.0 of the program.  Sad to report that it still seems like my privilege escalation issue for the "netsh" command still exists.  I would suggest that for the installation documentation for this software to add that **admin permissions are required** for proper crc cluster setup. I do not see this explicitly mentioned anywhere in the installation requirements (https://code-ready.github.io/crc/#installation_gsg).  It would be a good red flag for anyone working on corporate machines that do not have admin permissions on their machines (such as myself).

**Comment by jlynchMicron** [comment]
On second inspection, it looks like there is some mention of the need for admin permissions, just in the linux "sudo" terms, not windows "admin" terms.  When reading this the first time, I thought the sudo only applied to within the crc VM.

"Prerequisites
Your user account has permission to use the **sudo** command.

Do not run the crc binary as root (or Administrator). Always run the crc binary with your user account.

If you are setting up a new version, capture any changes made to the virtual machine before setting up a new CodeReady Containers release."

**Comment by robin-owen** [comment]
@jlynchMicron Thanks for pointing that out -- we could definitely be more clear there that `sudo` applies for Linux and macOS. The ability to escalate to administrator privileges is required on Windows. (I'm not sure how we should phrase this using the terminology of Windows.)

**Comment by robin-owen** [comment]
@gbraad Are Administrator privileges only required on Windows for `crc start`? I don't recall needing `sudo` permissions on Linux when starting the VM.

**Comment by robin-owen** [comment]
Fixed by #1457.

**Comment by jlynchMicron** [comment]
Sorry for edits on this issue.  Had to remove some information with "### redacted ###" due to security concerns.
